### ISSUE TITLE
False positive F811 when using TYPE_CHECKING

### ISSUE THREAD

**Issue by phil65** [astral-sh/ruff#5561]
For following code
```py
from typing import TYPE_CHECKING:
...
if TYPE_CHECKING:
    from x import y

...

if __name__ == "__main__":
    from x import y
    ...
```
Error F811 gets reported, which should not be the case I think.
Thank you for this great tool!

**Comment by zanieb** [comment]
Thanks for the issue! This indeed looks incorrect. Since this is just a small example, I'm curious why you're using this pattern in the real world?

Regardless, it seems like `from x import y` should be allowed in a new scope if the previous import was in an if-type-checking block.

You would expect the following to get reported still, right?

```python
from typing import TYPE_CHECKING:

if TYPE_CHECKING:
    from x import y

from x import y
```

**Comment by phil65** [comment]
Yes, I would expect that to get reported.
I am often using the `if __name__ == "__main__":` section for testing purposes, that´s why I am using this pattern.

**Comment by charliermarsh** [comment]
This no longer gets triggered (in the next version).
